controllers/k8ssandra/datacenters.go

```diff
@@ -434,3 +435,32 @@ func newRebuildTask(targetDc, namespace, srcDc string, numNodes int) *cassctlapi
 
 	return task
 }
+
+// sortDatacentersByPriority sorts the datacenters by upgrade priority.
+// The datacenters are sorted in descending order of priority.
+// The datacenters with the highest priority are first in the list.
+// The datacenters with the lowest priority are last in the list.
+func sortDatacentersByPriority(datacenters []api.CassandraDatacenterTemplate) []api.CassandraDatacenterTemplate {
+	sort.Slice(datacenters, func(i, j int) bool {
```

Comment:
`sort.Slice` performs an in-place sort and that's going to be a problem because this means that calling `sortDatacentersByPriority(kc.Spec.Cassandra.Datacenters)` will modify the input `kc.Spec.Cassandra.Datacenters`. You should instead create a copy of the slice inside the function.

Reply:
No, that's not how Golang slices work. You pass as value the reference to the slice. Even if you would do ``datacenters := datacenters``, you would only create a copy of the reference - not the underlying data.

https://go.dev/doc/effective_go#slices

So if you wish to copy the entire slice's data, you need to use copy() function.

```golang
datCenters := make([]type, len(datacenters))
copy(datacenters, datCenters)
```